### PR TITLE
upkeep[monorepo]: add changeset info to README

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Adds a brief section to the primary README that explains how to create a changeset.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,10 +42,24 @@ as well as a docker-compose file for bringing up local testnets easily
...
+### Changesets
+
+We use [changesets](https://github.com/atlassian/changesets) to manage releases of our various packages.
+You *must* include a `changeset` file in your PR when making a change that would require a new package release.
+
+Adding a `changeset` file is easy:
+
+1. Navigate to the root of the monorepo.
+2. Run `npx changeset`. You'll be prompted to select packages to include in the changeset. Use the arrow keys to move the cursor up and down, hit the `spacebar` to select a package, and hit `enter` to confirm your selection. Select *all* packages that require a new release as a result of your PR.
+4. Once you hit `enter` you'll be prompted to decide whether your selected packages need a `major`, `minor`, or `patch` release. Use a `major` release when making large changes that are not backwards-compatible. If you're unsure if you should use a `major` release, you probably want to use a `minor` or `patch` release. Use a `minor` release whenever you change the API of a package. Use a `patch` release when fixing bugs or otherwise making small changes that remain backwards-compatible.
```
Comment (tynes) [inline README.md:59]: You should like to this: https://semver.org/ and state that we follow semantic versioning

File: README.md [hunk 2]
```diff
@@ -42,10 +42,24 @@ as well as a docker-compose file for bringing up local testnets easily
...
+5. Commit your changeset and push it into your PR.
+6. Voilà, c'est fini!
```
Comment (tynes) [inline README.md:61]: The changeset bot on github will notice that a changeset is present in the PR

File: README.md [hunk 3]
```diff
@@ -42,10 +42,24 @@ as well as a docker-compose file for bringing up local testnets easily
...
 
+### Changesets
+
+We use [changesets](https://github.com/atlassian/changesets) to manage releases of our various packages.
+You *must* include a `changeset` file in your PR when making a change that would require a new package release.
+
+Adding a `changeset` file is easy:
+
+1. Navigate to the root of the monorepo.
+2. Run `npx changeset`. You'll be prompted to select packages to include in the changeset. Use the arrow keys to move the cursor up and down, hit the `spacebar` to select a package, and hit `enter` to confirm your selection. Select *all* packages that require a new release as a result of your PR.
```
Comment (maurelian) [inline README.md:58]: Should we stick with `yarn changeset` here just for consistency?

File: README.md [hunk 4]
```diff
@@ -42,10 +42,24 @@ as well as a docker-compose file for bringing up local testnets easily
...
+4. Once you hit `enter` you'll be prompted to decide whether your selected packages need a `major`, `minor`, or `patch` release. Use a `major` release when making large changes that are not backwards-compatible. If you're unsure if you should use a `major` release, you probably want to use a `minor` or `patch` release. Use a `minor` release whenever you change the API of a package. Use a `patch` release when fixing bugs or otherwise making small changes that remain backwards-compatible.
+5. Commit your changeset and push it into your PR.
```
Comment (maurelian) [inline README.md:60]: In the name of atomicity, should we specify that you should include your changeset with the commit that makes the change?

Comment (smartcontracts) [reply README.md:60]: For now I'm not going to include this. We should add some new docs that explain how to rebase and have a clean commit history.

**Review by maurelian — APPROVED** [review]

**Pull request merged**